### PR TITLE
Add documentation for changes to database & API

### LINKED ISSUE #1399: Document an end-to-end change
Choosing one of the past PRs that includes a DB migration and update to the API and documenting that process.

### PR DESCRIPTION
**Pull request by hursey013**
Fixes #1399 

This outlines the steps necessary to persist data on the frontend, database, and API.

### CODE REVIEW

**Comment by hursey013** [comment]
@ryanhofdotgov in regards to modifying the test scenarios (when newly added data needs to be represented) - are there any restrictions when changing those files that I should note in the documentation?

File: docs/updating-api-db.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Update XML template and test data
+
+After finding the corresponding XML file in `/api/templates`, make sure that the data is properly represented within the XML. It is also important to make sure that the new input is added to the corresponding `.json` file found within `/api/testdata` which is used to test the generated XML. Lastly, it's also ideal to add the new input to at least one of the complete scenarios in `/api/testdata/complete-scenarios`. This will allow the data to be pre-loaded and tested using the [load-scenario](https://github.com/18F/e-QIP-prototype/blob/develop/docs/test-scenarios.md#loading-existing-test-json-files) script.
```
Comment (ryanhofdotgov) [inline docs/updating-api-db.md]: @hursey013 One small recommended change to the guidance. 

Each one of the existing test cases 1-9 was created for OPM for a specific purpose – those are documented in `api/xml/xml_test.go`. Test case 6 was created for rejection (i.e, they weren't looking for specific areas of content) so it is a good test case to extend. I would leave the others as is. Alternatively, you can create a brand new test case.

**Comment by gidjin** [comment]
Thanks the documentation looks good to me, I'm glad you included example links. I'm not marking approval as Ryan had some good additional feedback.

File: docs/updating-api-db.md (hunk 1 above)
Comment (hursey013) [reply docs/updating-api-db.md]: That makes sense, just updated to add some clarification.

**Pull request merged**